src/Hosting/Hosting/src/Internal/HostingEventSource.cs

```diff
@@ -35,13 +55,16 @@ public void HostStop()
         [Event(3, Level = EventLevel.Informational)]
         public void RequestStart(string method, string path)
         {
+            Interlocked.Increment(ref _totalRequests);
```

Comment:
This is cool stuff. Is it worth investigating using `ThreadLocal<int>` and summing the `.Values` in the counter delegate? That would save a bit of time in the hot path by not locking the processor cores. Though I'm probably optimizing a bit prematurely ;)

Reply:
Yea I had a version with that though I'll do it after we measure this which I haven't since it involves turning on logging/eventsource which has other overhead. We also still need to interlocked.increment/decrement current requests which also made me go meh.

Reply:
Measure first is best